contracts/contracts/GrantRoundManager.sol

```diff
@@ -136,30 +137,22 @@ contract GrantRoundManager {
       _tokenIn.safeTransferFrom(msg.sender, _payoutAddress, _amountOut); // transfer funds directly to payout address
     } else {
       // Swap setup
-      uint24 _fee = _donation.fee;
-      uint256 _deadline = _donation.deadline;
-      uint256 _amountOutMinimum = _donation.amountOutMinimum;
-      uint160 _sqrtPriceLimitX96 = _donation.sqrtPriceLimitX96;
-
-      ISwapRouter.ExactInputSingleParams memory params = ISwapRouter.ExactInputSingleParams(
-        address(_tokenIn), // tokenIn
-        address(donationToken), // tokenOut
-        _fee, // fee
+      ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams(
+        _donation.path,
         _payoutAddress, // recipient
-        _deadline, // deadline
-        _amountIn, // amountIn
-        _amountOutMinimum, // amountOutMinimum
-        _sqrtPriceLimitX96 // sqrtPriceLimitX96
+        _donation.deadline,
+        _amountIn,
+        _donation.amountOutMinimum
       );
 
       // If user is sending a token, transfer it to this contract and approve the router to spend it
       if (msg.value == 0) {
-        _tokenIn.safeTransferFrom(msg.sender, address(this), _amountIn);
-        _tokenIn.approve(address(router), type(uint256).max); // TODO optimize so we don't call this every time
+        IERC20(_tokenIn).safeTransferFrom(msg.sender, address(this), _amountIn);
+        IERC20(_tokenIn).approve(address(router), type(uint256).max); // TODO optimize so we don't call this every time
```

Comment:
Is this case needed ? we do this already on L109

https://github.com/dcgtc/dgrants/blob/08817d4e38fe3c85cbb2b92d193306ec025a0572/contracts/contracts/GrantRoundManager.sol#L109

Reply:
Good catch, at first I made some changes that required this, then removed them and forgot to get rid of this. Fixed in ed764701e4f064c93f2b6036e292895a3cf2f315